openai-assistant/src/com/axonivy/connector/openai/assistant/ChatGptRequest.java

```diff
@@ -42,9 +42,19 @@ public String getModels() {
   }
 
   public String ask(String context, String question) {
+    return sendRequest(context, question, false);
+  }
+
+  public String insert(String context, String question) {
+    return sendRequest(context, question, true);
+  }
+
+  private String sendRequest(String context, String question, boolean includeSystemPrompt) {
     WebTarget chat = client.get().path("chat/completions");
     ArrayNode arrayNode = JsonNodeFactory.instance.arrayNode();
-    arrayNode.add(message("system", SYSTEM_PROMPT));
+    if (includeSystemPrompt) {
+      arrayNode.add(message("system", SYSTEM_PROMPT));
+    }
     arrayNode.add(message("user", String.format("%s \n\n %s", context, question)));
     ObjectNode request = completion().set("messages", arrayNode);
```

Comment:
Why no test on this rest-centered feature which is easy to re-produce in junit?

Reply:
Hi @ivy-rew 
I would handle what @Linh is working on from now. We are so sorry for this delay due to some critical issue regarding to marketplace website.
For the testing, we found no function from openai-assistant were tested.
Could we create a separated test project for open-ai-assistant or include it to the recent project?

Reply:
HI @ntqdinh-axonivy  
Sorry this didn't reach me. Please make sure you'll use the common team chat if you receive no answer for concrete requests. Otherwise request get lost simply by me being on vacation or something alike.

Answering the question however, there are already rest-centric tests which could easily be extended: https://github.com/axonivy-market/openai-connector/blob/master/openai-connector-test/src_test/com/axonivy/connector/openai/test/AiAssistanceTest.java
... also the utility method in here could be refactored and tested very simply within the existing testing project. So I wouldn't introduce a new test-project.

Reply:
Thanks for your hint @ivy-rew 
I will update the test cases for these changes soon.